cfme/tests/networks/nuage/test_operations.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+from wrapanapi.utils.random import random_name
+
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+from cfme.networks.provider.nuage import NuageProvider
+
+pytestmark = [
+    pytest.mark.provider([NuageProvider])
+]
+
+
+def test_router_add_subnet(provider, with_nuage_sandbox):
+    """
+    Ensure that subnet is added on network router
+
+    We navigate to router through Provider > Tenant > Network Router
+    """
+    sandbox = with_nuage_sandbox
+    tenant_name = sandbox['enterprise'].name
+    router_name = sandbox['domain'].name
+    tenant = provider.collections.cloud_tenants.instantiate(name=tenant_name, provider=provider)
+    router = tenant.collections.routers.instantiate(name=router_name)
+    subnet_name = random_name()
+    router.add_subnet(subnet_name, '100.100.100.0', '255.255.255.0', '100.100.100.1')
+    subnet = get_object_from_db_with_timeout(provider.appliance, 'cloud_subnets', subnet_name)
+
+    assert subnet is not None
```

Comment:
I think this test should be doing a few additional things to make it more complete in its validation.

`router.add_subnet` should (in my opinion) be instantiating and returning a Subnet entity instance.  You should be able to validate its existence by calling its `exists` method (which should navigate to its details page, through the Router entity).

The test should also be validating that the record pulled from the DB has a connection to the network router entity, not that its a generic subnet.

Reply:
I added the database check which validates that a subnet has a proper network_router. I also added subnet instance to the return statement of `router.add_subnet`. But do we really need to call `exists` method, shouldn't database check be enough?

Reply:
I'll leave that up to you, but validation that the entity is in the DB is not the same as validation that the entity is displayed correctly in the UI, and can be navigated to through the NetworkRouter.

Reply:
Hmm, you are right..., i think adding one extra assert won't hurt anyone :)